docs/reference/types.md

```diff
@@ -0,0 +1,81 @@
+(types)=
+# Type format
+
+SQLair supports structs maps and slices in input expressions and structs and
+maps in output expressions. 
+
+All types used with SQLair must be named so they can be referenced in the query.
+
+## Structs
+
+To use a struct with SQLair, all struct fields that correspond to a database
+column must be tagged with a `db` tag containing the column name. These fields
+must be public (i.e. start with a capital letter), though the struct itself does
+not have to be. Any fields without a tag will be ignored.
+
+If a struct contains an embedded struct then SQLair will treat the fields of the
+embedded structs as if they were fields in the parent struct.
+
+For example:
+```go
+type Person struct {
+    Name    string  `db:"name"`
+    ID      int     `db:"id_number"`
+    DoB     string  `db:"date_of_birth"`
+    Weight  int
+    height  int
+}
+```
+In this example the `Weight` and `height` fields will be ignored by SQLair.
+
+### The "omitempty" keyword
+
+The `omitempty` keyword tells SQLair to omit a column specified in a struct tag
+from an insert operation if the value in the corresponding struct field zeroed.
+
+This is useful if the content of the column is generated by the database, e.g.
+if it has an auto-increment directive.
+
+To add the `omitempty` keyword, write it in the `db` tag after the column name.
+
+For example:
+```go
+type Address struct {
+    ID   string `db:"id, omitempty"`
+    City string `db:"city"`
+}
+```
+
+> See more: {ref}`insert-statements`
+
+## Maps
+
+Named maps can be used with SQLair and must have a key with a base type of
+`string`. This is because SQLair uses the column names as keys when reading from
+and writing to the map. The value type of the map can be anything.
+
+For example:
+```go
+type Cols map[string]int
+```
+
+### sqlair.M
+For convenience, SQLair provides a named map type
+[`sqlair.M`](https://pkg.go.dev/github.com/canonical/sqlair#M) which has the
+type `map[string]any`.
+
+## Slices
+
+Named slices can only be used as inputs and will expand into a comma separated
```

Comment:
comma-separated* (I know some people don't dash on purpose, e.g., in physics journals where doing that would end up in dashes all over the place, but here I'd say it's not too much and feels appropriate).